File: thriftySwerve/src/main/java/frc/robot/commands/drive/TeleopDriveCommand.java
package frc.robot.commands.drive;

import edu.wpi.first.math.MathUtil;
import edu.wpi.first.wpilibj.Joystick;
import edu.wpi.first.wpilibj2.command.CommandBase;
import frc.robot.Constants.DriveConstants;
import frc.robot.subsystems.DriveSubsystem;

public class TeleopDriveCommand extends CommandBase {
  private final Joystick driveJoystick;
  private final DriveSubsystem driveSubsystem;

  public TeleopDriveCommand(Joystick driveJoystick, DriveSubsystem driveSubsystem) {
    this.driveJoystick = driveJoystick;
    this.driveSubsystem = driveSubsystem;
    addRequirements(driveSubsystem);
  }

  @Override
  public void execute() {
    driveSubsystem.drive(
        MathUtil.applyDeadband(driveJoystick.getX(), DriveConstants.kDeadbandAllStick)
            * driveSubsystem.getMaxSpeed(),
        MathUtil.applyDeadband(driveJoystick.getY(), DriveConstants.kDeadbandAllStick)
            * driveSubsystem.getMaxSpeed(),
        MathUtil.applyDeadband(driveJoystick.getTwist(), DriveConstants.kDeadbandAllStick)
            * driveSubsystem.getMaxOmega());
  }

  @Override
  public boolean isFinished() {
    return false;
  }

  @Override
  public void end(boolean interrupted) {
    driveSubsystem.drive(0, 0, 0);
  }
}

File: jif/src/main/kotlin/org/strykeforce/command/TeleOpDriveCommand.kt
package org.strykeforce.command

import edu.wpi.first.wpilibj.command.Command
import org.strykeforce.robotComponents
import org.strykeforce.thirdcoast.swerve.SwerveDrive.DriveMode.TELEOP

const val DEADBAND = 0.08

class TeleOpDriveCommand : Command() {

    val drive = robotComponents.driveSubsystem()
    val controls = robotComponents.controls()

    init {
        requires(drive)
    }

    override fun initialize() {
        drive.driveMode = TELEOP
    }

    override fun execute() {
        val forward = controls.forward.applyDeadband(DEADBAND)
        val strafe = controls.strafe.applyDeadband(DEADBAND)
        val azimuth = controls.azimuth.applyDeadband(DEADBAND)
        drive.drive(forward, strafe, azimuth)
    }

    override fun isFinished() = false
}

private fun Double.applyDeadband(amount: Double) = if (Math.abs(this) < amount) 0.0 else this

